File: Java/binarysearch.java
//This program performs binary search on the given array.

package Java_Samples;
import java.util.Scanner;

public class binarysearch {
    public static void main(String []args){
        Scanner sc = new Scanner(System.in);
        int a[] = {10,9,8,7,4,3,2};
        int ub = 6;
        int lb = 0;
        System.out.println ("Enter the element to search:");
        int key = sc.nextInt();
        sc.close();
        while(lb <= ub){
            int mid = (ub+lb)/2;
            if(key == a[mid]){ 
                System.out.println("Element found at index:"+ mid); 
                return;
            }
            else if(key < a[mid]){
                lb = mid+1;
            }
            else{
                ub = mid-1;
            }
        }
        System.out.println("Element not found!");
        sc.close();
    }
}

File: binarysearch.py
#This program performs binary search on the given array.

a = [10,20,30,40,50,80]
lb = 0
ub = 5
key = int(input("Enter the element to search:"))
found = False
while(lb<=ub):
    m = (lb+ub)//2
    if(a[m]==key):
        print("Element found at position ",m)
        found = True
        break 
    elif(key<a[m]):
        ub = m-1
    else:
        lb = m+1
if not found:
    print("Element not found")
